tests/flows/relayer_modifies_message.go

```diff
@@ -12,23 +12,25 @@ import (
 	predicateutils "github.com/ava-labs/subnet-evm/predicate"
 	"github.com/ava-labs/subnet-evm/x/warp"
 	teleportermessenger "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/TeleporterMessenger"
-	"github.com/ava-labs/teleporter/tests/network"
+	"github.com/ava-labs/teleporter/tests/interfaces"
 	"github.com/ava-labs/teleporter/tests/utils"
-	localUtils "github.com/ava-labs/teleporter/tests/utils/local-network-utils"
 	gasUtils "github.com/ava-labs/teleporter/utils/gas-utils"
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/log"
 	. "github.com/onsi/gomega"
 )
 
+type constructSignedMessageFunc func(
+	ctx context.Context,
+	sourceReceipt *types.Receipt,
+	source interfaces.SubnetTestInfo,
+	destination interfaces.SubnetTestInfo,
+) []byte
+
 // Disallow this test from being run on anything but a local network, since it requires special behavior by the relayer
-func RelayerModifiesMessage() {
-	network := &network.LocalNetwork{}
-	subnets := network.GetSubnetsInfo()
-	Expect(len(subnets)).Should(BeNumerically(">=", 2))
-	subnetAInfo := subnets[0]
-	subnetBInfo := subnets[1]
+func RelayerModifiesMessage(network interfaces.Network, constructSignedMessageFunc constructSignedMessageFunc) {
```

Comment:
For tests that should only be run on local networks, what do you think about modifying the signature to accept a `localNetwork` rather than a `network`?

Reply:
I thought of doing that, but unfortunately it creates circular imports between the `local` and `flows` packages.

Another option that I'm open to switching to would be to add the two functions currently passed as parameters to flows to the `network` interface, and then have the implementation of them always error for networks they are not supported on. Thoughts on that?

Reply:
That would work, though I'm not a huge fan of adding functions that are only used in one place to an interface.

One other alternative suggestion: What if we added a `LocalNetwork` interface to `package interfaces`? It could just be composed of a `Network`, but that should allow Go's type system to make the distinction. Then we change this function to accept a `LocalNetwork`.

Lots of ways to skin a cat ofc, I don't have too strong an opinion either way since this is all just for testing.

Reply:
I like even better, good call. Updated it now.

Also I've never heard "lots of way to skin a cat" before, but I like it haha 😅

Reply:
Haha definitely one of the odder idioms